subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -306,14 +352,30 @@ static psa_status_t cracen_signature_ecc_sign(int message, const psa_key_attribu
 	sign.r = (char *)signature;
 	sign.s = (char *)signature + *signature_length / 2;
 
-	if (message) {
+	/* ED25519PH requires prehashing and supports sign and verify message
+	 * the message is therefore hashed here before si_sig_verify is called
+	 */
+	if (alg == PSA_ALG_ED25519PH && message) {
+		uint8_t status;
+		uint8_t hash[64];
+		size_t output_len;
+
+		status = psa_hash_compute(PSA_ALG_SHA_512, input,
```

Comment:
We should probably return an error if status is not success here, right?

Reply:
Yes that makes sense. I have added in a return of the status from psa_hash_compute if it is not a success. Technically that would break psa spec as psa_hash_compute has some returns that verify_message does not. However it would only be the case if the length, size of hash or alg were wrong. All of which set in the code directly. Could replace it with a specific return code, but as there are no return codes that directly implies a hash failure that seems less user friendly